services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -4322,6 +4336,20 @@ void preCheckStorePartitionCountUpdate(String clusterName, Store store, int newP
     }
   }
 
+  private void updateRealTimeTopicName(Store store) {
+    String oldRealTimeTopicName = Utils.getRealTimeTopicName(store);
+    String newRealTimeTopicName;
+    PubSubTopic newRealTimeTopic;
+
+    do {
+      newRealTimeTopicName = Utils.createNewRealTimeTopicName(oldRealTimeTopicName);
+      newRealTimeTopic = getPubSubTopicRepository().getTopic(newRealTimeTopicName);
+      oldRealTimeTopicName = newRealTimeTopicName;
+    } while (getTopicManager().containsTopic(newRealTimeTopic));
```

Comment:
Who's expected to create this new real time topic?

Reply:
RealTimeTopicSwitcher I believe. This piece is just storing the name that is to be used the next time rt topic is created, whomsoever.

Reply:
So are we going to block update store command until topic name is created?

Reply:
yes. is `containsTopic` that expensive? I am expecting only one call of this.